catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/CreateOperations.java

```diff
@@ -349,11 +249,21 @@ public CreateResponse create(CreateStorageRequest streamCreateRequest)
 
             createResponse = create(createRequest);
         } catch (Exception e) {
-            opsCrudSupport.handleStorageException(createStorageRequest,
-                    streamCreateRequest.getId(),
-                    e);
+            if (createStorageRequest != null) {
+                try {
+                    sourceOperations.getStorage()
+                            .rollback(createStorageRequest);
+                } catch (StorageException e1) {
+                    LOGGER.error("Unable to remove temporary content for id: {}",
+                            createStorageRequest.getId(),
+                            e1);
+                }
+            }
+            throw new IngestException(
+                    "Unable to store products for request: " + streamCreateRequest.getId(), e);
+
         } finally {
-            opsCrudSupport.commitAndCleanup(createStorageRequest,
+            opsStorageSupport.commitAndCleanup(createStorageRequest,
```

Comment:
At first glance i'm not sure this should be in a finally this seems weird that we would throw an exception above and then do this commitAndCleanup.  That being said this has nothing to do with this PR it looks like its always been this way.